pkg/operator/hive/hive.go

```diff
@@ -86,9 +86,7 @@ func (r *ReconcileHiveConfig) deployHive(hLog log.FieldLogger, h *resource.Helpe
 		hiveContainer.Env = append(hiveContainer.Env, syncsetReapplyIntervalEnvVar)
 	}
 
-	if len(instance.Spec.ManagedDomains) > 0 {
-		addManagedDomainsVolume(&hiveDeployment.Spec.Template.Spec)
-	}
+	addManagedDomainsVolume(&hiveDeployment.Spec.Template.Spec, mdConfigMap.Name)
```

Comment:
Is there a technical reason why we had to change this behavior? It seems strange to me to have an empty configmap.

Reply:
i though it made the code easier to always assume there will be a configmap. i originally was preserving the behavior of not using a configmap in the absence of any managed domains, and just was dealing with a lot of special case if-no-managed-domains code. I can restore the old behavior if the empty configmap is really that surprising.

Reply:
OK. If it trickles down to a lot of places where you need to deal with the special case, then there is not too much harm in having an empty configmap.